src/context.cpp

```diff
@@ -155,34 +147,35 @@ context_t::remove(const std::string& name) {
     return service;
 }
 
-boost::optional<const actor_t&>
+boost::optional<quote_t>
 context_t::locate(const std::string& name) const {
-    auto ptr = m_services.synchronize();
-    auto it  = std::find_if(ptr->begin(), ptr->end(), match{name});
+    return m_services.apply([&](const service_list_t& list) -> boost::optional<quote_t> {
+        auto it = std::find_if(list.begin(), list.end(), match{name});
 
-    if(it == ptr->end()) {
-        return boost::none;
-    }
+        if (it != list.end()) {
+            auto  endpoints = it->second->endpoints();
+            auto& prototype = it->second->prototype();
 
-    return boost::optional<const actor_t&>(it->second->is_active(), *it->second);
-}
+            // TODO(@kobolog): Figure out if there should always be some endpoints for a
+            // service. Useless is_active() check was dropped, so adding this assert JIC.
+            BOOST_ASSERT(!endpoints.empty());
```

Comment:
It shouldn't be empty in all cases I THINK, hence assert. I'm going to refactor actor_t, so I keep it here to assert the expected variant.